File: _3_b_algorithms_in_python/_1_recursion_and_backtracking/_1_basics/_5_array_sum.py

#  sum of elements upto index n-1 of array a of length n
def sum_of_array(n, a):
    if n < 0:
        return 0
    return sum_of_array(n - 1, a) + a[n]

# Example usage:
n = int(input("Enter the size of the array: "))
a = [int(input()) for _ in range(n)]
result = sum_of_array(n - 1, a)
print(result)

File: _8_recursion_and_backtracking/_1_basics/_5_array_sum.cpp
#include <iostream>
using namespace std;

// sum of elements upto index n-1 of array a of length n

int sumOfArray(int n, int a[])
{
    if (n < 0)
        return 0;
    return sumOfArray(n - 1, a) + a[n];
}

int main()
{
    int n;
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
    }
    cout << sumOfArray(n - 1, a);
}